### PR TITLE
executor: make prepare analyze killed globally instead of query quota

### LINKED ISSUE #44320: Prepared "analyze table" statement cancelled due to exceeds "tidb_mem_quota_query"
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Use playground to build a tidb server
2. Use "~/.tiup/bin/tiup bench tpch prepare -D tpch1 -T 4 -H 127.0.0.1 -P 4000 --sf 1" to prepare data
3. run "set tidb_mem_quota_query = 50000000;" to set tidb_mem_quota_query to a small value
4. run prepare stmt from "analyze table lineitem";
5. run execute stmt;

### 2. What did you expect to see? (Required)
The statement executes successfully.
### 3. What did you see instead (Required)
ERROR 1105 (HY000): Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again.[conn=1833862249828057511]
### 4. What is your TiDB version? (Required)

| Release Version: v7.2.0-alpha
Edition: Community
Git Commit Hash: 65d5ef2eaa34c17e73928d17be339b8dbdcddc11
Git Branch: heads/refs/tags/v7.2.0-alpha
UTC Build Time: 2023-05-31 02:32:04
GoVersion: go1.20.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #44320 & #40946

Problem Summary:
For now, `prepare stmt from "analyze table t"; execute stmt; ` is not recognized as an analyze statement and is not managed by global memory quota instead of query memory quota.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/memory/tracker.go [hunk 1]
```diff
@@ -88,11 +88,11 @@ type Tracker struct {
...
 	// following fields are used with atomic operations, so make them 64-byte aligned.
-	bytesConsumed       int64            // Consumed bytes.
-	bytesReleased       int64            // Released bytes.
-	maxConsumed         atomicutil.Int64 // max number of bytes consumed during execution.
-	SessionID           uint64           // SessionID indicates the sessionID the tracker is bound.
-	NeedKill            atomic.Bool      // NeedKill indicates whether this session need kill because OOM
+	bytesConsumed       int64             // Consumed bytes.
+	bytesReleased       int64             // Released bytes.
+	maxConsumed         atomicutil.Int64  // max number of bytes consumed during execution.
+	SessionID           atomicutil.Uint64 // SessionID indicates the sessionID the tracker is bound.
```
Comment (chrysan) [inline util/memory/tracker.go:94]: This change is to fix data race. cc @XuHuaiyu

**Review by yibin87 — COMMENTED** [review]
LGTM，but not reviewer yet.

**Review by qw4990 — APPROVED** [review]

File: util/memory/tracker.go [hunk 2]
```diff
@@ -88,11 +88,11 @@ type Tracker struct {
...
+	NeedKill            atomic.Bool       // NeedKill indicates whether this session need kill because OOM
```
Comment (hawkingrei) [inline util/memory/tracker.go:95]: Here use ```sync/atomic``` and ```go.uber.org/atomic```. I think We can unify this.

Comment (chrysan) [reply util/memory/tracker.go:95]: This is for an urgent hotfix. What about doing it in another PR, including atomic in other files?

Comment (hawkingrei) [reply util/memory/tracker.go:95]: OK

**Review by hawkingrei — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**